src/blocks/mrc_call_python_function.ts

```diff
@@ -612,7 +676,18 @@ export const pythonFromBlock = function(
       const functionName = callPythonFunctionBlock.mrcActualFunctionName
           ? callPythonFunctionBlock.mrcActualFunctionName
           : block.getFieldValue(FIELD_FUNCTION_NAME);
-      code = 'self.' + componentName + '.' + functionName;
+      // Generate the correct code depending on the module type.
+      switch (generator.getModuleType()) {
+        case commonStorage.MODULE_TYPE_PROJECT:
+          code = 'self.';
+          break;
+        case commonStorage.MODULE_TYPE_MECHANISM:
```

Comment:
I don't think this is correct.   The mechanism won't have access to the robot, so I think the mechanism module needs to be with the project one. (which we still need to rename to robot)